deploy/crds/build.dev_buildruns_crd.yaml

```diff
@@ -280,6 +280,35 @@ spec:
                     required:
                     - url
                     type: object
+                  sources:
```

Comment:
Could you make `source` optional to encourage users to move to `sources` ?

Reply:
@sbose78 Why do you need to make `source` be optional? 

By going through [`remote-artifacts`](https://github.com/otaviof/build/blob/a34984aa02182180dfe091b651dec913ec99eaa0/docs/proposals/remote-artifacts.md) proposal, my understanding about `.spec.sources` field is it's just an **alternative dependencies** during building image. 

For instance, now we want to build this [source code](https://github.com/otaviof/nodejs-ex), however, I don't want to use the default [logo](https://github.com/otaviof/nodejs-ex/blob/master/assets/images/logo.png) in this repo. Then I can specify an [alternative logo path](https://gist.githubusercontent.com/otaviof/cf16a90d8364777a2ac927bce0332f16/raw/5888a860ca5e65f9367c3210bb05fcf6538928ea/logo.png) via `.spec.sources` filed.

Then when we create a buildRun, our build controller will prepend a step to download this alternative logo before building image. And this default logo will be covered with this alternative logo in the workspace. Next start to build image.

Not sure if my understanding is correct or not? @otaviof

Reply:
We might be too early to discuss how to make `.spec.source` optional, it does need more work and attention to the API than it's involved in the Remote Artifacts initial implementation.

Here we are adding new features to the operator, and later on we can concentrate on how the CRD/API will look like, in a subsequent PR.

Reply:
Thanks @otaviof ! 
I see. But I also want to confirm my understanding above is correct or not:
>my understanding about .spec.sources field is it's just an alternative dependencies during building image.
>
If the field `.spec.sources` is used for dependencies of `.spec.source` or not?

Reply:
@xiujuan95 you're welcome! :-) So, on this PR we are keeping the original behavior of `.spec.source`, and plus, adding more features with `.spec.sources`. However, there are plans to combine both of those fields, we should touch this subject during community meetings in the near future.

Reply:
I see, thanks @otaviof !